joern-cli/frontends/c2cpg/src/main/scala/io/joern/c2cpg/astcreation/AstForTypesCreator.scala

```diff
@@ -30,16 +31,30 @@ trait AstForTypesCreator { this: AstCreator =>
     Ast(cpgNamespace)
   }
 
+  private def typeForIASTDeclarator(
+    declaration: IASTSimpleDeclaration,
+    declarator: IASTDeclarator,
+    index: Int
+  ): String = {
+    declarator match {
+      case arrayDecl: IASTArrayDeclarator => registerType(cleanType(typeFor(arrayDecl)))
+      case _ =>
+        safeGetBinding(declarator.getName) match {
+          case Some(variable: ICPPVariable) if variable.getType.isInstanceOf[CPPClosureType] =>
+            registerType(Defines.Function)
+          case _ =>
+            registerType(cleanType(typeForDeclSpecifier(declaration.getDeclSpecifier, index = index)))
+        }
+    }
+  }
+
   protected def astForDeclarator(declaration: IASTSimpleDeclaration, declarator: IASTDeclarator, index: Int): Ast = {
     val name = shortName(declarator)
     declaration match {
       case d if isTypeDef(d) && shortName(d.getDeclSpecifier).nonEmpty =>
         val filename = fileName(declaration)
-        val typeDefName = if (name.isEmpty) {
-          safeGetBinding(declarator.getName).map(b => registerType(b.getName))
-        } else {
-          Option(registerType(name))
-        }
+        val typeDefName = if (name.isEmpty) { safeGetBinding(declarator.getName).map(b => registerType(b.getName)) }
+        else { Option(registerType(name)) }
```

Comment:
Is this a result of auto-formatting. Personally I like the original version more, or doing the if/else on 2 lines as it is here, but getting rid of the `{}`. Just a minor style nit